source/fuzz/protobufs/spvtoolsfuzz.proto

```diff
@@ -381,6 +381,55 @@ message SideEffectWrapperInfo {
   uint32 value_to_copy_id = 7;
 }
 
+message ReturnMergingInfo {
+  // TransformationMergeFunctionReturns needs to modify each merge block of
+  // loops containing return instructions, by:
+  // - adding instructions to decide whether the function is returning
+  // - adding instructions to pass on the return value of the function,
+  //   if it is returning
+  // - changing the branch instruction (which must be an unconditional branch)
+  //   to a conditional branch that, if the function is returning, branches to
+  //   the merge block of the innermost loop that contains this merge block
+  //   (which can be the new merge block introduced by the transformation).
+  //
+  // One such merge block of the form:
+  //  %block = OpLabel
+  //   %phi1 = OpPhi %type1 %val1_1 %pred1 %val1_2 %pred2
+  //   %phi2 = OpPhi %type2 %val2_1 %pred1 %val2_2 %pred2
+  //           OpBranch %next
+  //
+  // is transformed into:
+  //               %block = OpLabel
+  //     %is_returning_id = OpPhi %bool %false %pred1 %false %pred2 %is_bb1_returning %ret_bb1 %true %is_bb2_returning
```

Comment:
I think something is wrong here.  I guess the new predecessors should be %ret_bb1 and %ret_bb2, and the associated values should be %is_bb1_returning and %is_bb2_returning - i.e. %true should not be there?

Reply:
There should actually be a %true when the corresponding predecessor was originally a merge block. That line was still wrong, though. I fixed it and made a distinction between predecessors that where return blocks and merge block predecessors that might be returning